### DISCUSSION TITLE
Reader does not recognize variables with valid names

### DISCUSSION THREAD

**Discussion by madeira-ag** [OpenDrift/opendrift#913] [General]
Hello!

I'm just starting with Opendrift and found this error that I cannot go around. My netCDF has the variables u, v, um and vm, but when I read it with reader_netCDF_CF_generic, I get the message:

Skipped variables without standard_name: ['u', 'v', 'um', 'vm']

I changed the variables names with ncf4 package on R to "x_sea_water_velocity" or "sea_water_x_velocity" (and the same for y) but I keep getting the same message (this time like Skipped variables without standard_name: ['sea_water_x_velocity', 'sea_water_y_velocity', 'um', 'vm']), witch leads me to think that, for some reason, variables with valid names are being seen as not named correctly. What can I do? Is this the right reader for my file? Is something wrong with the file itself?

Thank you for your attention!

**Comment by knutfrode** [comment]
Hi,

It is not the variable names that shall be `x_sea_water_velocity` etc, but rather their attribute called `standard_name`.
So the variable names can be anything (it is ignored).
So you could add the corresponding `standard_name` attributes to the variables in your files, but OpenDrift also contains a mechanism to "map" your variables to the correct `standard_name` without needing to modify the original files.

This is illustrated in this example:
https://opendrift.github.io/gallery/example_macondo.html
For your case, the mapping would be like `....'u': 'x_sea_water_velocity', ....`

**Reply by madeira-ag** [reply]
Thank you so much for the quick reply!

I tried and it worked very well. I've seen this notation ('u':'x_sea_water_velocity') before, but didnt know where to use it. I had no knowledge of this standard_name concept. Thank you again!
